packages/flutter/lib/src/rendering/sliver_padding.dart

```diff
@@ -162,16 +162,14 @@ class RenderSliverPadding extends RenderSliver with RenderObjectWithChildMixin<R
       to: mainAxisPadding + childLayoutGeometry.scrollExtent,
     );
     final double mainAxisPaddingPaintExtent = beforePaddingPaintExtent + afterPaddingPaintExtent;
+    final double paintExtent = math.min(
+      beforePaddingPaintExtent + math.max(childLayoutGeometry.paintExtent, childLayoutGeometry.layoutExtent + afterPaddingPaintExtent),
+      constraints.remainingPaintExtent,
+    );
```

Comment:
Previously, the layoutExtent would occasionally (due to floating point math) end up slightly bigger than the paintExtent. That prevents that from happening. I don't know how to actually cause it to happen. It was intermittently happening on that flickering dropdown test you saw me look at.

I'm happy to add a test if you have any ideas for how to actually reproduce the problem reliably.